src/components/pages/User/UserInformationInput.tsx

```diff
@@ -0,0 +1,142 @@
+import error_svg from "../../../assets/images/icons/material.io/error-24px.svg";
+import {biggerMaxStrLength, notMinStrLength} from "../../../background/methods/checkInput";
+import {deleteSpaces} from "../../../background/methods/strings";
+import {DEFAULT_ALERT_DURATION, MAX_PASSWORD_LENGTH, MIN_PASSWORD_LENGTH} from "../../../background/constants";
+
+export interface UserInformationInputInterface {
+    username: string,
+    password: string,
+    passwordConfirmation?: string
+}
+
+type UserInformationInputProps = {
+    triggerAlert(duration: number, color: "primary" | "secondary" | "success" | "danger" | "warning" | "info" | "light" | "dark", message: string): void,
+    submitFunction(newUser: UserInformationInputInterface): void,
+    presets?: UserInformationInputInterface
+}
+
+export default function UserInformationInput(props: UserInformationInputProps): ReactElement {
+    let {triggerAlert, submitFunction, presets} = props;
+    const [username, setUsername] = useState<string>(presets?.username??"");
+    const [password, setPassword] = useState<string>(presets?.password??"");
+    const [passwordConfirmation, setPasswordConfirmation] = useState<string>(presets?.password??"");
+    const [passwordInformationLength, setPasswordInformationLength] = useState<boolean>(false);
+    const [passwordInformationLowercase, setPasswordInformationLowercase] = useState<boolean>(false);
+    const [passwordInformationUppercase, setPasswordInformationUppercase] = useState<boolean>(false);
+    const [passwordInformationNumber, setPasswordInformationNumber] = useState<boolean>(false);
+    const [passwordsMatch, setPasswordsMatch] = useState<boolean>(true);
+
+    const reviewPasswordMatch = useCallback((): void => {
+        setPasswordsMatch(password === passwordConfirmation);
+    }, [password, passwordConfirmation]);
+
+    useEffect(() => {
+        reviewPasswordMatch()
+    }, [reviewPasswordMatch])
+
+    const makePasswordInputFitRules = (input: string): [string, boolean] => {
+        input = deleteSpaces(input);
+        if (biggerMaxStrLength(input, MAX_PASSWORD_LENGTH)) {
+            triggerAlert(DEFAULT_ALERT_DURATION, "warning", "Maximum password length exceeded. Input was undone.");
+            return [input, false];
+        }
+        return [input, true];
+    }
+
+    const handlePasswordChange = (event: ChangeEvent<HTMLInputElement>) => {
+        event.preventDefault();
+        let value: [string, boolean] | string = makePasswordInputFitRules(event.target.value);
+        if (!value[1]) {
+            value = password;
+        } else {
+            value = value[0]
+        }
+        setPasswordInformationLength(!notMinStrLength(value, MIN_PASSWORD_LENGTH));
+        setPasswordInformationLowercase(value.match(/[a-z]/) !== null);
+        setPasswordInformationUppercase(value.match(/[A-Z]/) !== null);
+        setPasswordInformationNumber(value.match(/\d/) !== null);
```

Comment:
redundant, this code should only be defined once

Reply:
it is connected to some css, will talk about this next time